### ISSUE TITLE
weird configuration error

### ISSUE THREAD

**Issue by osa1** [haskell/cabal#2518]
```
➜  chsc git:(master) ✗ cabal install --allow-newer
Resolving dependencies...
Notice: installing into a sandbox located at
/home/omer/haskell/chsc/.cabal-sandbox
Configuring graph-wrapper-0.2.4.3...
Configuring lattices-1.2.1.1...
Configuring prettyclass-1.0.0.0...
Configuring random-1.1...
Configuring text-1.2.0.4...
Failed to install graph-wrapper-0.2.4.3
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/graph-wrapper-0.2.4.3.log ):
Failed to install random-1.1
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/random-1.1.log ):
Failed to install prettyclass-1.0.0.0
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/prettyclass-1.0.0.0.log ):
Failed to install lattices-1.2.1.1
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/lattices-1.2.1.1.log ):
Failed to install text-1.2.0.4
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/text-1.2.0.4.log ):
cabal: Error: some packages failed to install:
cpphs-1.18.9 depends on text-1.2.0.4 which failed to install.
graph-wrapper-0.2.4.3 failed during the configure step. The exception was:
user error (The package 'graph-wrapper' requires Cabal library version
>=1.19.2 && >=1.2 but no suitable version is installed.)
haskell-src-exts-1.16.0.1 depends on text-1.2.0.4 which failed to install.
lattices-1.2.1.1 failed during the configure step. The exception was:
user error (The package 'lattices' requires Cabal library version >=1.19.2 &&
>=1.2 but no suitable version is installed.)
polyparse-1.11 depends on text-1.2.0.4 which failed to install.
prettyclass-1.0.0.0 failed during the configure step. The exception was:
user error (The package 'prettyclass' requires Cabal library version >=1.19.2
&& >=0 but no suitable version is installed.)
random-1.1 failed during the configure step. The exception was:
user error (The package 'random' requires Cabal library version >=1.19.2 &&
>=1.8 but no suitable version is installed.)
supercompile-0.1 depends on text-1.2.0.4 which failed to install.
text-1.2.0.4 failed during the configure step. The exception was:
user error (The package 'text' requires Cabal library version >=1.19.2 &&
>=1.8 but no suitable version is installed.)
```

However, I can install dependencies manually like this:

```
➜  chsc git:(master) ✗ cabal install lattices-1.2.1.1
Resolving dependencies...
Notice: installing into a sandbox located at
/home/omer/haskell/chsc/.cabal-sandbox
Configuring lattices-1.2.1.1...
Building lattices-1.2.1.1...
Installed lattices-1.2.1.1
...
```

But then it fails with this:

```
➜  chsc git:(master) ✗ cabal install --allow-newer
Resolving dependencies...
Notice: installing into a sandbox located at
/home/omer/haskell/chsc/.cabal-sandbox
Configuring supercompile-0.1...
Failed to install supercompile-0.1
Build log ( /home/omer/haskell/chsc/.cabal-sandbox/logs/supercompile-0.1.log ):
cabal: Error: some packages failed to install:
supercompile-0.1 failed during the configure step. The exception was:
user error (The package 'supercompile' requires Cabal library version >=1.19.2
&& >=1.2 but no suitable version is installed.)
```

The version I'm using should just work:

```
➜  chsc git:(master) ✗ cabal --version
cabal-install version 1.22.0.1
using version 1.22.1.1 of the Cabal library
```

**Comment by 23Skidoo** [comment]
Duplicate of #2023. Try `cabal install Cabal`. The error message should really be improved.
